backend/src/index.js

```diff
@@ -320,22 +310,48 @@ async function main() {
   setTimeout(regularCheckNodeStatus, 0);
 
   // Periodic check of validator's fee and delegators
-  const regularCheckValidatorsExtraInfo = async (validators) => {
-    for (let i = 0; i < validators.length; i++) {
-      const { account_id } = validators[i];
-      validators[i].fee = await nearRpc.callViewMethod(
-        account_id,
-        "get_reward_fee_fraction",
-        {}
-      );
-      validators[i].delegators = await nearRpc.callViewMethod(
-        account_id,
-        "get_number_of_accounts",
-        {}
-      );
+  const regularCheckValidatorsExtraInfo = async () => {
+    if (validators) {
+      validatorsExtraInfo = validatorsExtraInfo || validators;
+
+      for (let i = 0; i < validatorsExtraInfo.length; i++) {
+        const { account_id } = validatorsExtraInfo[i];
+        validatorsExtraInfo[i].fee = await nearRpc.callViewMethod(
+          account_id,
+          "get_reward_fee_fraction",
+          {}
+        );
+        validatorsExtraInfo[i].delegators = await nearRpc.callViewMethod(
+          account_id,
+          "get_number_of_accounts",
+          {}
+        );
+      }
     }
-    return validators;
+
+    if (proposals) {
+      proposalsExtraInfo = proposalsExtraInfo || proposals;
+
+      for (let i = 0; i < proposalsExtraInfo.length; i++) {
+        const { account_id } = proposalsExtraInfo[i];
+        proposalsExtraInfo[i].fee = await nearRpc.callViewMethod(
+          account_id,
+          "get_reward_fee_fraction",
+          {}
+        );
+        proposalsExtraInfo[i].delegators = await nearRpc.callViewMethod(
+          account_id,
+          "get_number_of_accounts",
+          {}
+        );
+      }
+    }
```

Comment:
@frol , I don't know is it a good idea or not... Could you please look at it and tell me your mind?